### PR TITLE
CLDR-14268 improve vetting participation page

### PR DESCRIPTION
**Pull request by srl295**
CLDR-14268

- major rewrite of 'vetting participation' to a speedy query + new UI (non JSP)
- also fix Utilka CLDR-6320
- also cleanup a leaking Statement from UserRegistry

### CODE REVIEW

**Comment by srl295** [comment]
The participation page has moved, and is now under the gear (next to Forum participation).  *There's now no reason to toggle on or off 'show participation' because it is so quick— seconds.*

Image for review:
- "no coverage" notes that the organization has this locale on their list, but no vetters are assigned
- bold indicates assigned vetters
- locales with no voting is marked off

image

**Comment by srl295** [comment]
Whoops. Java issue, maybe source level ?

**Review by macchiati — DISMISSED** [review]
Looks good to me, but Tom should also review

**Comment by srl295** [comment]
ahh…  `java.util.Set.of(…)` is a JDK9 feature.  @yumaoka @macchiati should we consider someday upping our JDK level? I think I bumped It from 6 to 7 not too long ago!

File: tools/cldr-apps/src/main/webapp/js/CldrSurveyVettingLoader.js [hunk 1]
```diff
@@ -641,141 +641,146 @@ function showV() {
...
-								display: surveyUserPerms.userExist
-							},
-
-							{
-								divider: true
-							},
-							{
-								title: 'My Votes'
-							}, // My Votes section
+							divider: true
```
Comment (btangmu) [inline tools/cldr-apps/src/main/webapp/js/CldrSurveyVettingLoader.js:644]: Hard to see what changed other than whitespace...

File: tools/cldr-apps/src/main/webapp/js/CldrSurveyVettingLoader.js [hunk 2]
```diff
@@ -641,141 +641,146 @@ function showV() {
...
+							level: 2,
+							url: surveyUserURL.manageUser,
+							display: (surveyUserPerms.userIsTC || surveyUserPerms.userIsVetter)
+						},
+						{
+							special: 'forum_participation', /* Cf. special_forum_participation */
+							level: 2,
+							display: surveyUserPerms.userCanMonitorForum
+						},
+						{
```
Comment (btangmu) [inline tools/cldr-apps/src/main/webapp/js/CldrSurveyVettingLoader.js:723]: is this block for vetting_participation the only non-whitespace change in this function?

File: tools/cldr-apps/src/main/webapp/js/special/vetting_participation.js [hunk 3]
```diff
@@ -0,0 +1,208 @@
...
+		+ ")";
+
+
+
+	function Page() {
+		// constructor
+	}
+	
+	/**
+	 * Fetch the Forum Participation data from the server, and "load" it
```
Comment (btangmu) [inline tools/cldr-apps/src/main/webapp/js/special/vetting_participation.js:30]: "Vetting" not "Forum"

File: tools/cldr-apps/src/main/webapp/js/special/vetting_participation.js [hunk 4]
```diff
@@ -0,0 +1,208 @@
...
+	 *
+	 * @param params an object with various properties; see SpecialPage.js
+	 */
+	function show(params) {
```
Comment (btangmu) [inline tools/cldr-apps/src/main/webapp/js/special/vetting_participation.js:34]: show() is over 100 lines, way more than one screenful -- could improve with subroutines

**Review by btangmu — DISMISSED** [review]
I suggested a few improvements which could be done in a subsequent PR

**Comment by srl295** [comment]
Adding this also:

image

File: tools/cldr-apps/src/main/java/org/unicode/cldr/web/SurveyVettingParticipation.java [hunk 5]
```diff
@@ -0,0 +1,175 @@
...
+    boolean showCodes = false;
+
+    public SurveyVettingParticipation(String org, SurveyMain sm) {
+        this.sm = sm;
+        this.org = org;
+        this.missingLocalesForOrg = org; // TODO
+    }
+
+    public void getJson(JSONWriter r) throws SQLException, JSONException {
+       final StandardCodes sc = StandardCodes.make();
```
[the remark this answers is not in the document]
Comment (srl295) [reply tools/cldr-apps/src/main/java/org/unicode/cldr/web/SurveyVettingParticipation.java:40]: done

File: tools/cldr-apps/src/main/webapp/js/CldrSurveyVettingLoader.js (hunk 2 above)
Comment (srl295) [reply tools/cldr-apps/src/main/webapp/js/CldrSurveyVettingLoader.js:723]: I don't know what happened with the whitespace here. Can you hit the Gear and review with 'ignore whitespace'?

File: tools/cldr-apps/src/main/webapp/js/CldrSurveyVettingLoader.js [hunk 6]
```diff
@@ -641,141 +641,146 @@ function showV() {
...
+							special: 'vetting_participation', /* Cf. special_vetting_participation */
+							level: 2,
+							display: surveyUserPerms.userIsTC || surveyUserPerms.userIsVetter
+						},
+						{
+							title: 'LOCKED: Note: your account is currently locked.',
+							level: 2,
+							display: (surveyUserPerms.userIsLocked || false),
```
Comment (srl295) [inline tools/cldr-apps/src/main/webapp/js/CldrSurveyVettingLoader.js:731]: undefined was being interpreted as true (probably comparing to `! == false`

File: tools/cldr-apps/src/main/webapp/js/special/vetting_participation.js (hunk 4 above)
Comment (srl295) [reply tools/cldr-apps/src/main/webapp/js/special/vetting_participation.js:25]: fixed

**Comment by srl295** [comment]
rebased

**Comment by macchiati** [comment]
I need to check what our limit is.

**Comment by srl295** [comment]
> I need to check what our limit is.
> […](#)

I've already fixed this to work on Java 7

**Review by macchiati — APPROVED** [review]

**Pull request merged**